[Zonnie-23] iP

Zonnie-23 opened the pull request:
### Nova
> "Your mind is for having ideas, not holding time." - David Allen [source](https://dansilvestre.com/productivity-quotes)

Nova frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ *SUPER FAST* to use

All you need to do is,
1. download it from [here](https://github.com/Zonnie-23/ip/releases).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Zonnie-23 committed:
Renamed to Nova and added greetings

Zonnie-23 committed:
add feature: Echo

Zonnie-23 committed:
add feature: Add, List

Zonnie-23 committed:
add feature: Mark as Done

Zonnie-23 committed:
add feature: New Tasks: Event, Deadline and Todo

Zonnie-23 committed:
added automated UI testing

Zonnie-23 committed:
Modified code to use exceptions

Zonnie-23 committed:
added feature: delete

Zonnie-23 committed:
Added saving feature

Zonnie-23 committed:
Merge branch 'branch-Level-7'

Zonnie-23 committed:
Add load and save feature

Zonnie-23 committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Add load and save feature

Zonnie-23 committed:
Refactor code

Zonnie-23 committed:
Refactor code: DeleteCommand class added

Zonnie-23 committed:
Implement Date-Time feature

Zonnie-23 committed:
Update test cases

Zonnie-23 committed:
Add TaskList.java

Zonnie-23 committed:
Merge branch 'branch-A-MoreOOP'

Zonnie-23 committed:
Add Todo.java

Zonnie-23 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Zonnie-23 committed:
Add JUnit test

Zonnie-23 committed:
Merge branch 'branch-A-JUnit'

Zonnie-23 committed:
Add JavaDoc

Zonnie-23 committed:
Update to follow coding conventions

Zonnie-23 committed:
Resolve conflict

Zonnie-23 committed:
Update Ui package name

Zonnie-23 committed:
Add find feature

Zonnie-23 committed:
Merge branch 'branch-Level-9'

yjunyeu commented on src/main/java/Command/ByeCommand.java:
```diff
@@ -0,0 +1,62 @@
+import Ui.Ui;
+
+public class ByeCommand implements Command {
+    private final TaskList toDoList;
+    private final Ui ui;
+    private final Storage storage;
+    private final Scanner scanner;
```
Should these constant names be made all uppercase?

Zonnie-23 replied:
Thanks for your comments. I believe constants are static final variables that are deeply immutable. Therefore, I feel that these are not constants and thus do not need to be all uppercase. Same for the rest.

yjunyeu replied:
I see. Apologies for the oversight.

yjunyeu commented on src/main/java/Command/DeadlineCommand.java:
```diff
@@ -0,0 +1,58 @@
+ * Command to add a deadline task to the task list
+ */
+public class DeadlineCommand implements Command {
+    private final TaskList toDoList;
+    private final Ui ui;
+    private final String[] msgParts;
+    private final LocalDateTime deadlineTime;
```
Should these constant names be made all uppercase?

yjunyeu commented on src/main/java/Command/EventCommand.java:
```diff
@@ -0,0 +1,60 @@
+public class EventCommand implements Command {
+    private TaskList toDoList;
+    private Ui ui;
+    private String[] instruction;
```
Should a collection have a plural form?

yjunyeu commented on src/main/java/Command/FindCommand.java:
```diff
@@ -0,0 +1,59 @@
+ * </p>
+ */
+public class FindCommand implements Command {
+    private final TaskList toDoList;
+    private final String keyword;
+    private final Ui ui;
+    private final TaskList matchedTaskList = new TaskList();
```
Should these constant names be made all uppercase?

yjunyeu commented on src/main/java/Command/StatusUpdateCommand.java:
```diff
@@ -0,0 +1,75 @@
+    private TaskList toDoList;
+    private Ui ui;
+    private int taskIndex;
+    private boolean isDone;
```
I like how you made this boolean variable sound boolean. Good job!

yjunyeu commented on src/main/java/Storage/Storage.java:
```diff
@@ -0,0 +1,121 @@
+ * Handles file-based storage operations for tasks.
+ */
+public class Storage {
+    private final File dataFile;
```
Should the constant name be made all uppercase?

yjunyeu commented on src/main/java/Task/Deadline.java:
```diff
@@ -0,0 +1,52 @@
+ * Represent a task with a deadline.
+ */
+public class Deadline extends Task {
+    private final LocalDateTime dueTime;
```
Should the constant name be made all uppercase?

yjunyeu reviewed (commented):
Overall code looks really good and clean. The OOP was well thought out as well.
Just some nits on coding standard violations

Zonnie-23 committed:
Rename package to follow convention

Zonnie-23 committed:
Resolve coding violations

Zonnie-23 committed:
Merge branch 'branch-A-CheckStyle'

Zonnie-23 committed:
Add basic GUI

Zonnie-23 committed:
Add proper resizing features

Zonnie-23 committed:
Merge branch 'branch-Level-10'

Zonnie-23 committed:
Add Assertions

Zonnie-23 committed:
Add separate save feature

Save command: allow user to save todo list at any time

User can only save when exiting program

User may accidentally enter no when exiting the programs. Therefore, to avoid such situations, allow

Let's handle another case where the user responds save to the chatbot. In this case, the current To-do List will be saved. When exiting the program, the program will only save before exit if the user responds "save" when prompted.

This is to minimise accidental loss of unsaved data, while not overaggressively saving/updating user's list.

Zonnie-23 committed:
Refactor Nova.java

The read method in the Nova class currently does not obey SLAP.

SLAP makes code easier to read, and also reduces duplicate code

Let's add new command classes Help and Exit. Also, some of the
operations in read has been split into a seperate method parseCommand.

This is to ensure obey SLAP and to keep code clean.

Zonnie-23 committed:
Merge pull request #1 from Zonnie-23/branch-A-Assertions

Add Assertions

Zonnie-23 committed:
Merge branch 'master' into branch-A-CodeQuality

Zonnie-23 committed:
Merge pull request #2 from Zonnie-23/branch-A-CodeQuality

Merge code quality improvements

Zonnie-23 committed:
Add feature to display schedule on given day

Zonnie-23 committed:
Update README.md

Zonnie-23 committed:
Schedule tasks in order

Zonnie-23 committed:
Improve GUI and Update README.md

Zonnie-23 committed:
Update code to comply with codestyle

Zonnie-23 committed:
Adjust code to comply with convention
